### PR TITLE
Generate markdown table

### PR DESCRIPTION
**Pull request by burnettk**
script to generate markdown table

## Summary by CodeRabbit

- **New Features**
  - Introduced a script to generate Markdown tables from column information and data.

- **Tests**
  - Added test cases to validate the Markdown table generation, including handling of special characters like vertical bars.

### CODE REVIEW

File: spiffworkflow-backend/src/spiffworkflow_backend/scripts/generate_markdown_table.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+from spiffworkflow_backend.models.script_attributes_context import ScriptAttributesContext
+from spiffworkflow_backend.services.jinja_service import JinjaHelpers
+from spiffworkflow_backend.scripts.script import Script
+
+
+class GenerateMarkdownTable(Script):
+    @staticmethod
+    def requires_privileged_permissions() -> bool:
+        """We have deemed this function safe to run without elevated permissions."""
+        return False
```
[the remark this answers is not in the document]
Comment (burnettk) [reply spiffworkflow-backend/src/spiffworkflow_backend/scripts/generate_markdown_table.py:12]: it's a boolean function. false means doesn't need privileged.

File: spiffworkflow-backend/tests/spiffworkflow_backend/scripts/test_generate_markdown_table.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            columns=columns,
+            data=data,
+        )
+        expected_result = (
+            "| Name | Description |\n"
+            "| ---- | ---- |\n"
+            "| Alice | Alice's description \\| with a vertical bar |\n"
+            "| Bob | Bob's description \\| with another vertical bar |\n"
+        )
+        assert result == expected_result
```
[the remark this answers is not in the document]
Comment (burnettk) [reply spiffworkflow-backend/tests/spiffworkflow_backend/scripts/test_generate_markdown_table.py:67]: it's tested by an integration test.

File: spiffworkflow-backend/src/spiffworkflow_backend/scripts/generate_markdown_table.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    def get_description(self) -> str:
+        return """Given column info and data, returns a string suitable for use in markdown to show a table."""
+
+    def normalize_table_headers_to_dicts(self, columns: list) -> list[dict]:
+        return [{"property": col, "label": col.replace("_", " ").capitalize()} if not isinstance(col, dict) else col for col in columns]
```
[the remark this answers is not in the document]
Comment (burnettk) [reply spiffworkflow-backend/src/spiffworkflow_backend/scripts/generate_markdown_table.py]: we have lint running in ci, so it's not possible for the line length to exceed the app-configured limit when those pass, which they are.

**Comment by madhurrya** [comment]
@burnettk Is there anything to test in this? If so how?

**Comment by burnettk** [comment]
see video for how to use `generate_markdown_table(columns, data)`. let me know if i can clarify anything.

**Pull request merged**